text/0000-near-shell.md

```diff
@@ -0,0 +1,449 @@
+
+# Motivation
+[motivation]: #motivation
+
+These enhancements are intended to simplify and enhance developer and user productivity by providing a single command-line tool which exercises all major features of NEAR. As each major piece of NEAR functionality is added to `near-shell`, we should see utilization of each of those features increase.  Additionally, community mindshare and understanding of NEAR should increase as `near-shell` will contain detailed help documentation for each major feature with clear examples of use, just like Linux man pages and/or other major CLI tools which have integrated help for each command.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This NEP will jump straight into the concepts necessary to understand the components needed in `near-shell`, and generally an extensible CLI tool. Unlike other NEPs, the guide-level and reference-level explanations are combined. There is nothing overly technical in the following sections, but more conceptual in nature. With that, the discussion begins by separating the three types of stored information needed for a successful, long-term `near-shell`.
+
+## Settings, config, and key management
+
+Shell-experience **settings**, connection **configuration**, and **key management** are the three types of stored data necessary for developers to reliably build and deploy projects.
+ 
+```
+─ awesome-near-project  ⟵ NEAR dApp
+ ├── .near-config       ⟵ Stores shell-experience settings and connection configuration
+ │  ├── connections     ⟵ Contains files used to connect and deploy a contract (except keys)
+ │  │  ├── default.js   ⟵ Default configuration is for development
+ │  │  └── localnet.js  ⟵ Custom connection added by user for localnet
+ │  └── settings.js     ⟵ Stores near-shell settings, how shell behaves when run in this project
+ └── .near-credentials  ⟵ Previously "neardev" this directory contains private key inforamtion
+    └── default         
+       └── alice.json   ⟵ NEAR account "alice" has private key information here
+```
+
+### Definitions:
+
+#### 1. Settings
+[shell-experience-settings]: #shell-experience-settings
+
+---
+
+These are key-values reflecting how `near-shell` behaves when called within a NEAR project.
+    
+Example: a developer using OS X uses `near login` with access to a browser, whereas a validator runs `near login` on a CentOS box with no UI or browser. A user prompt may ask "Is this login from a computer with a browser?" The answer is then stored in key-value format in the project-level directory so it can be referenced later, skipping the prompt in the future.
+
+Besides answers to user prompts, shell-experience settings also store information about the last version of `near-shell` used in this project. For more information, please see [Upgradability](#upgradability).
```

Comment:
This seems like work for `package.json` file not for a `.<something>` folder. Usually `.` folders are added to gitignore as they contain local configuration.

https://stackoverflow.com/questions/10065564/add-custom-metadata-or-config-to-package-json-is-it-valid